tests/contrib/flask/test_flask_appsec.py

```diff
@@ -235,3 +235,21 @@ def test_flask_body_json_empty_body_logs_warning(self):
             self._aux_appsec_prepare_tracer()
             self.client.post("/", data="", content_type="application/json")
             assert "Failed to parse werkzeug request body" in self._caplog.text
+
+    def test_flask_body_json_bad_logs_warning(self):
+        with self._caplog.at_level(logging.WARNING), override_global_config(dict(_appsec_enabled=True)):
+            self._aux_appsec_prepare_tracer()
+            self.client.post("/", data="not valid json", content_type="application/json")
+            assert "Failed to parse werkzeug request body" in self._caplog.text
+
+    def test_flask_body_xml_bad_logs_warning(self):
+        with self._caplog.at_level(logging.WARNING), override_global_config(dict(_appsec_enabled=True)):
+            self._aux_appsec_prepare_tracer()
+            self.client.post("/", data="bad xml", content_type="application/xml")
+            assert "Failed to parse werkzeug request body" in self._caplog.text
+
+    def test_flask_body_xml_empty_logs_warning(self):
+        with self._caplog.at_level(logging.WARNING), override_global_config(dict(_appsec_enabled=True)):
+            self._aux_appsec_prepare_tracer()
+            self.client.post("/", data="bad xml", content_type="application/xml")
+            assert "Failed to parse werkzeug request body" in self._caplog.text
```

Comment:
What is the difference between these two? they look identical to me

Reply:
Good catch, second one should have an empty strings, fixed.